[Chua Shi Hong] ip

chuashihong opened the pull request:
# Duke Chatbot
> "Hello I'm Duke, a newborn ChatBot" --- Duke
Duke Chatbot is a task tracking bot that helps you in your daily life, it is:

- text-based
- have emotion
- ~~quite~~ VERY stupid (still learning)

What you need to do is,
1. download him [here](https://github.com/chuashihong/ip/releases)
2. open it
3. try to add your first task: todo first task
4. hurrray!

##### And what's best? It is **FREE**

Features:
 
- [x] Manage Task
- [x] There are 3 types of task: Todo task, Deadline task and Event task 
- [x] Keep track of the date
- [ ] Beautiful UI with chatbot (coming soon)

Commit:
Add Gradle support

chuashihong committed:
Level 1: "bye" feature, and echo feature

chuashihong committed:
Level 2 and Level 3

chuashihong committed:
Level-4 Complete

chuashihong committed:
A-TextUiTesting

chuashihong committed:
Level-5 completed: will try to include more exception to make chatbot more powerful :)

chuashihong committed:
Level-5

chuashihong committed:
Level-6: delete function

chuashihong committed:
handled more exception

chuashihong committed:
Added more test case for automated test

chuashihong committed:
Add more exception handling case

chuashihong committed:
Level-7 Save

chuashihong committed:
level-8 do nothing yet

chuashihong committed:
Merge branch 'branch-Level-8'

chuashihong committed:
Merge branch 'branch-Level-7'

# Conflicts:
#	src/main/java/Duke.java

chuashihong committed:
Merge branch 'master' into branch-Level-8

chuashihong committed:
Level-8 Date

chuashihong committed:
Merge commit '9223c5d1991a5c04e0e9b7eadeb011996eeb2931'

chuashihong committed:
Add package, more OOP(except parser)

chuashihong committed:
Add JavaDoc

chuashihong committed:
add Junit test

chuashihong committed:
Merge branch 'branch-Level-8'

# Conflicts:
#	data/tasks.txt
#	src/main/java/Duke.java
#	src/main/java/duke/Deadline.java
#	src/main/java/duke/Event.java

chuashihong committed:
Merge branch 'master' into branch-A-JavaDoc

# Conflicts:
#	data/tasks.txt
#	src/main/java/duke/Deadline.java
#	src/main/java/duke/Event.java

chuashihong committed:
Merge branch 'branch-A-JavaDoc'

chuashihong committed:
Refactor: Command Package, Parser, JavaDoc, Ui

chuashihong committed:
Commands, more JavaDoc, JAR file, Junit test

chuashihong committed:
Level-9

chuashihong committed:
Change packages name

lowkaiwei98 commented on src/main/java/duke/Storage.java:
```diff
@@ -0,0 +1,104 @@
+ * Storage class that helps to save or load existing data from local site
+ */
+public class Storage {
+    private final String filePath;
```
Shouldn't constant be in uppercase and underline or i think final shouldn't be use if it is not a constant

lowkaiwei98 commented on src/main/java/duke/Ui.java:
```diff
@@ -0,0 +1,163 @@
+    private final String lines = "____________________________" +
+            "________________________________";
+    private final String endline = "___________________________" +
+            "_________________________________\n";
```
Variable name for constant should be in uppercase? i noticed similar issue in other areas

lowkaiwei98 commented on src/main/java/duke/command/DeadlineCommand.java:
```diff
@@ -0,0 +1,43 @@
+
+import java.io.IOException;
+
+public class DeadlineCommand extends Command {
```
Every public class should have a comment even if it inherits from abstract class, i recommend writing 😄

lowkaiwei98 commented on src/main/java/duke/command/EventCommand.java:
```diff
@@ -0,0 +1,44 @@
+package duke.command;
+
+
+import duke.*;
```
wildcard should not be use in imports?

lowkaiwei98 reviewed (commented):
Overall, i found your code looks good to read, just have to fix some nits 😄

justinyjt commented on data/tasks.txt:
```diff
@@ -0,0 +1,5 @@
+T | 0 | key chua shionh
+T | 0 | asdf keyasshole
```
nice 👍

justinyjt commented on src/main/java/duke/Ui.java:
```diff
@@ -0,0 +1,163 @@
+     * Tells the user that the task is not done yet
+     * @param markTask current list of tasks
+     */
+    public void showMarkTaskAsAlreadyDone(Task markTask) {
```
I like how you give very detailed method names to explain what the method does 👍

bloodofme replied:
Great JavaDocs done here 😄

justinyjt commented on src/main/java/duke/task/Deadline.java:
```diff
@@ -0,0 +1,43 @@
+ */
+public class Deadline extends Task {
+
+    protected String by;
```
Perhaps this variable name could be slightly more descriptive? Something like "doByDate"? 😅

justinyjt commented on src/main/java/duke/task/Deadline.java:
```diff
@@ -0,0 +1,43 @@
+     * @return A formatted string
+     */
+    @Override
+    public String dataFormatOfTask() {
```
Maybe this method name could be rephrased to start with a verb like "formatTaskData"?

justinyjt reviewed (commented):
Overall, I feel that your code quality is very good and there are no major violations of the code quality guidelines. Keep up the great work 👍

chuashihong committed:
Fix some issues about Date

chuashihong committed:
Merge remote-tracking branch 'origin/add-gradle-support'

bloodofme commented on src/main/java/duke/Storage.java:
```diff
@@ -0,0 +1,102 @@
+import duke.task.Task;
+import duke.task.Todo;
+
+import java.io.*;
```
Based on the coding standard, perhaps you should list out all imported classes explicitly 🤔

bloodofme commented on src/main/java/duke/Storage.java:
```diff
@@ -0,0 +1,102 @@
+     */
+    public void saveFile(TaskList taskList) throws IOException {
+        FileWriter fileWriter = new FileWriter(filePath);
+        for (int i = 0; i <taskList.size() ; i++) {
```
Based on coding standards, it may be better to add a spacing before `<`, and remove the spacing after `;`
`i < taskList.size();`

bloodofme reviewed (commented):
Overall, just some minor violations of coding standards. Good Code!

chuashihong committed:
Added GUI in Duke

chuashihong committed:
Refactor the code so that it follows the coding standard

Wrote a GUI for Duke

The GUI can accept the user input
And the Duke will echo the respond from user input
More responses will be added later

Add GUI to create a better user experience

chuashihong committed:
Test if the branch-Level-10 appear in my local computer
A dummy test

chuashihong committed:
Refactor ui class, parser and Command class and command variant classes.

-ui method now return string
-Parser:parseCommand now returns string
-Command.execute now returns string
-GUI now properly return response to user

chuashihong committed:
Added enum class Action

-Action can handle various user input as a command
-Commit more files that the previous commit had missed

chuashihong committed:
Merge branch 'branch-Level-10'

chuashihong committed:
Add an extension to detect duplicated task

Definition of duplicated task
-a task is the same type(todo, deadlline, event)
-a task has the same description
-a task has been on the list before

Also refactor the name of the variable in Event, Todo, and Deadline class
- to make it easier to understand the variable

chuashihong committed:
Merge branch 'C-DetectDuplicates'

chuashihong committed:
Update ui class so that the list command shows the task correctly

chuashihong committed:
Revert "Update ui class so that the list command shows the task correctly"

This reverts commit ca4503c41622327b2428fb999433287289d09388.

chuashihong committed:
Add a CheckDuplicate class to initilize extension

chuashihong committed:
Merge branch 'master' of https://github.com/chuashihong/ip

chuashihong committed:
Added a assertion line for debugging purpose+

chuashihong committed:
Merge pull request #2 from chuashihong/C-DetectDuplicates

Added a assertion line for debugging purpose+

chuashihong committed:
Improve code quality by reviewing all the codes
-Extract some method to make it simpler to follow the steps

chuashihong committed:
Further improve code quality and also clear out some unused code on the files

chuashihong committed:
Update readme.md

chuashihong committed:
Update README.md

chuashihong committed:
Update README.md

chuashihong committed:
Update readme.md in docs
Update Ui.png which shows the screenshot of the GUI of Duke
Fix a bug where the response does not create new line correctly

chuashihong committed:
Merge branch 'master' of https://github.com/chuashihong/ip

# Conflicts: Fixed all the conflicts

#	docs/README.md

chuashihong committed:
Update readme,md

chuashihong committed:
Update README.md

chuashihong committed:
Update README.md

chuashihong committed:
Update README.md

chuashihong committed:
Duke now can quit the application automatically after pressing 'bye'

chuashihong committed:
Added an extension: Help command, to guide a user to use the application.

chuashihong committed:
Improve GUI to make it better
